metricflow/test/query/test_query_parser.py

```diff
@@ -584,3 +608,16 @@ def test_duplicate_metric_query(bookings_query_parser: MetricFlowQueryParser) ->
 def test_no_metrics_or_group_by(bookings_query_parser: MetricFlowQueryParser) -> None:  # noqa: D
     with pytest.raises(InvalidQueryException, match="no metrics or group by items"):
         bookings_query_parser.parse_and_validate_query()
+
+
+def test_offset_metric_with_diff_agg_time_dims_error() -> None:  # noqa: D
+    metrics_yaml_file = YamlConfigFile(filepath="inline_for_test_1", contents=METRICS_YAML)
+    revenue_yaml_file = YamlConfigFile(filepath="inline_for_test_1", contents=REVENUE_YAML)
+    query_parser = query_parser_from_yaml(
+        [EXAMPLE_PROJECT_CONFIGURATION_YAML_CONFIG_FILE, revenue_yaml_file, metrics_yaml_file]
+    )
+    with pytest.raises(InvalidQueryException, match="does not match any of the available group-by-items"):
```

Comment:
Out of curiosity, does this raise a more specific metric-time related error somewhere? Might be a nice additional improvement.